test/Microsoft.Dnx.Runtime.Tests/DependencyManagement/FrameworkReferenceResolverFacts.cs

```diff
@@ -0,0 +1,69 @@
+        [InlineData("net451", "mscorlib", @"REFASMSROOT\.NETFramework\v4.5.1\mscorlib.dll", "4.0.0.0")]
+        [InlineData("net452", "mscorlib", @"REFASMSROOT\.NETFramework\v4.5.2\mscorlib.dll", "4.0.0.0")]
+        [InlineData("net46", "mscorlib", @"REFASMSROOT\.NETFramework\v4.6\mscorlib.dll", "4.0.0.0")]
+
+        // System.Printing (added in 3.0)
+        [InlineData("net30", "System.Printing", @"REFASMSROOT\v3.0\System.Printing.dll", "3.0.0.0")]
+        [InlineData("net35", "System.Printing", @"REFASMSROOT\v3.0\System.Printing.dll", "3.0.0.0")]
+        [InlineData("net40", "System.Printing", @"REFASMSROOT\.NETFramework\v4.0\System.Printing.dll", "4.0.0.0")]
+        [InlineData("net45", "System.Printing", @"REFASMSROOT\.NETFramework\v4.5\System.Printing.dll", "4.0.0.0")]
+        [InlineData("net451", "System.Printing", @"REFASMSROOT\.NETFramework\v4.5.1\System.Printing.dll", "4.0.0.0")]
+        [InlineData("net452", "System.Printing", @"REFASMSROOT\.NETFramework\v4.5.2\System.Printing.dll", "4.0.0.0")]
+        [InlineData("net46", "System.Printing", @"REFASMSROOT\.NETFramework\v4.6\System.Printing.dll", "4.0.0.0")]
+
+        // System.Core (added in 3.5)
+        [InlineData("net35", "System.Core", @"REFASMSROOT\v3.5\System.Core.dll", "3.5.0.0")]
+        [InlineData("net40", "System.Core", @"REFASMSROOT\.NETFramework\v4.0\System.Core.dll", "4.0.0.0")]
+        [InlineData("net45", "System.Core", @"REFASMSROOT\.NETFramework\v4.5\System.Core.dll", "4.0.0.0")]
+        [InlineData("net451", "System.Core", @"REFASMSROOT\.NETFramework\v4.5.1\System.Core.dll", "4.0.0.0")]
+        [InlineData("net452", "System.Core", @"REFASMSROOT\.NETFramework\v4.5.2\System.Core.dll", "4.0.0.0")]
+        [InlineData("net46", "System.Core", @"REFASMSROOT\.NETFramework\v4.6\System.Core.dll", "4.0.0.0")]
+
+        // Microsoft.CSharp (added in 4.0)
+        [InlineData("net40", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.0\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net45", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.5\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net451", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.5.1\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net452", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.5.2\Microsoft.CSharp.dll", "4.0.0.0")]
+        [InlineData("net46", "Microsoft.CSharp", @"REFASMSROOT\.NETFramework\v4.6\Microsoft.CSharp.dll", "4.0.0.0")]
+
+        public void FrameworkResolverResolvesCorrectPaths(string shortFrameworkName, string assemblyName, string expectedPath, string expectedVersion)
+        {
+            var progFiles = string.IsNullOrEmpty(Environment.GetEnvironmentVariable("PROGRAMFILES(X86)")) ?
+                Environment.GetEnvironmentVariable("PROGRAMFILES") :
+                Environment.GetEnvironmentVariable("PROGRAMFILES(X86)");
+            var refAsmsRoot = Path.Combine(progFiles, "Reference Assemblies", "Microsoft", "Framework");
```

Comment:
Use this method https://github.com/aspnet/dnx/blob/dev/src/Microsoft.Dnx.Runtime/FrameworkReferenceResolver.cs#L141